src/configure/services/azureDevOpsService.ts

```diff
@@ -0,0 +1,200 @@
+            let parts = part.split("/");
+            return parts[2].trim();
+        }
+        else {
+            throw new Error("Repo Url is not of Azure Repos type.");
+        }
+    }
+
+    public async getRepositoryId(repositoryName: string, remoteUrl: string): Promise<string> {
+        this.setOrganizationAndProjectNameFromRepositoryUrl(remoteUrl);
+        let repositoryDetails = await this.azureDevOpsClient.getRepositoryDetails(repositoryName, this.organizationName, this.projectName);
+        return repositoryDetails.id;
+    }
+
+    public async listOrganizations(): Promise<string[]> {
+        let organizations = await this.azureDevOpsClient.listOrganizations();
+        let organizationNames: string[] = [];
+        for (let organization of organizations.value) {
+            organizationNames.push(organization.accountName);
+        }
+
+        return organizationNames;
+    }
+
+    public getOrganizationName(): string {
+        return this.organizationName;
+    }
+
+    /***
+     * This function will only set organization name if it is not already set.
+     * Once the value is set, it cannot be altered.
+     */
+    public setOrganizationName(organizationName: string): void {
+        if (!this.organizationName) {
+            this.organizationName = organizationName;
+        }
+    }
+
+    public async listProjects(): Promise<string[]> {
+        let projects = await this.azureDevOpsClient.listProjects(this.organizationName);
+        let items: string[] = [];
+        for (let project of projects.value) {
+            items.push(project.name);
+        }
+
+        return items;
+    }
+
+    public getProjectName(): string {
+        return this.projectName;
+    }
+
+    /***
+     * This function will only set organization name if it is not already set.
+     * Once the value is set, it cannot be altered.
+     */
+    public setProjectName(projectId: string): void {
+        if (!this.projectName) {
+            this.projectName = projectId;
+        }
+    }
+
+    public async createGitHubServiceConnection(gitHubPat: string, prefix: string): Promise<string> {
+        let endpointName: string = prefix.concat(guidGenerator().substr(0, 5));
+        let response = await this.azureDevOpsClient.createGitHubServiceConnection(endpointName, gitHubPat, this.organizationName, this.projectName);
+        let endpointId: string = response.id;
+        await this.waitForGitHubEndpointToBeReady(endpointId);
+        await this.azureDevOpsClient.authorizeEndpointForAllPipelines(endpointId, endpointName, this.organizationName, this.projectName)
+            .then((response) => {
+                if (response.allPipelines.authorized !== true) {
+                    throw new Error("Could not authorize endpoint for use in Pipelines.");
+                }
+            });
+
+        return endpointId;
+    }
+
+    public async analyzeRepoAndSuggestPipelines(repoDetails: GitRepositoryDetails) {
```

Comment:
is this added by mistake?

Reply:
Had added it as we plan to leverage repo analysis already implemented in azure-pipelines.

Reply:
But now I see it is causing confusion, thus deleting it.